homeassistant/components/vallox/sensor.py

```diff
@@ -0,0 +1,209 @@
+"""Support for Vallox ventilation unit sensors."""
+
+import logging
+
+from homeassistant.const import (
+    DEVICE_CLASS_HUMIDITY, DEVICE_CLASS_TEMPERATURE, TEMP_CELSIUS)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+
+from . import DOMAIN, METRIC_KEY_MODE, SIGNAL_VALLOX_STATE_UPDATE
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the sensors."""
+    if discovery_info is None:
+        return
+
+    name = hass.data[DOMAIN]['name']
+    state_proxy = hass.data[DOMAIN]['state_proxy']
+
+    sensors = [
+        ValloxProfileSensor(
+            name="{} Current Profile".format(name),
+            state_proxy=state_proxy,
+            device_class=None,
+            unit_of_measurement=None,
+            icon='mdi:gauge'
+
+        ),
+        ValloxFanSpeedSensor(
+            name="{} Fan Speed".format(name),
+            state_proxy=state_proxy,
+            metric_key='A_CYC_FAN_SPEED',
+            device_class=None,
+            unit_of_measurement='%',
+            icon='mdi:fan'
+        ),
+        ValloxSensor(
+            name="{} Extract Air".format(name),
+            state_proxy=state_proxy,
+            metric_key='A_CYC_TEMP_EXTRACT_AIR',
+            device_class=DEVICE_CLASS_TEMPERATURE,
+            unit_of_measurement=TEMP_CELSIUS,
+            icon=None
+        ),
+        ValloxSensor(
+            name="{} Exhaust Air".format(name),
+            state_proxy=state_proxy,
+            metric_key='A_CYC_TEMP_EXHAUST_AIR',
+            device_class=DEVICE_CLASS_TEMPERATURE,
+            unit_of_measurement=TEMP_CELSIUS,
+            icon=None
+        ),
+        ValloxSensor(
+            name="{} Outdoor Air".format(name),
+            state_proxy=state_proxy,
+            metric_key='A_CYC_TEMP_OUTDOOR_AIR',
+            device_class=DEVICE_CLASS_TEMPERATURE,
+            unit_of_measurement=TEMP_CELSIUS,
+            icon=None
+        ),
+        ValloxSensor(
+            name="{} Supply Air".format(name),
+            state_proxy=state_proxy,
+            metric_key='A_CYC_TEMP_SUPPLY_AIR',
+            device_class=DEVICE_CLASS_TEMPERATURE,
+            unit_of_measurement=TEMP_CELSIUS,
+            icon=None
+        ),
+        ValloxSensor(
+            name="{} Humidity".format(name),
+            state_proxy=state_proxy,
+            metric_key='A_CYC_RH_VALUE',
+            device_class=DEVICE_CLASS_HUMIDITY,
+            unit_of_measurement='%',
+            icon=None
+        ),
+        ValloxSensor(
+            name="{} Remaining Time For Filter".format(name),
+            state_proxy=state_proxy,
+            metric_key='A_CYC_REMAINING_TIME_FOR_FILTER',
+            device_class=None,
+            unit_of_measurement='days',
```

Comment:
We don't want to store state that is relative time. That generates not needed state updates. It's ok to use an absolute utc iso formatted timestamp and set the device class to timestamp. Then the absolute timestamp will show as a relative time in the frontend.

https://developers.home-assistant.io/docs/en/entity_sensor.html#available-device-classes